services/movie_session.py

```diff
@@ -0,0 +1,61 @@
+from datetime import datetime
+
+from db.models import MovieSession
+from django.db.models import QuerySet
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> MovieSession:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall_id=cinema_hall_id,
+        movie_id=movie_id
+    )
+
+
+def get_movies_sessions(
+        session_date: str = None
+) -> QuerySet:
+    if session_date:
+        return MovieSession.objects.filter(
+            show_time__date=session_date
+        )
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> str:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: str = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None
+) -> QuerySet:
+    sess = MovieSession.objects.all()
+
+    if show_time is not None:
+        sess.update(
+            show_time=show_time
+        )
+    if movie_id is not None:
+        sess.filter(id=session_id).update(

```

Comment:
you can simplify your code by retrieving the needed movie session object using .get() method before all the **if** statements and than assign to its attributes the needed values :

movie_session.movie_id = movie_id

and after all the **if** statements you can save all the changes using .save() method.

Reply:
Thank you))))))